contracts/database/orm/observer.go

```diff
@@ -1,30 +1,61 @@
 package orm
 
 type Observer interface {
-	// Retrieved called when the model is retrieved from the database.
-	Retrieved(Event) error
-	// Creating called when the model is being created.
-	Creating(Event) error
 	// Created called when the model has been created.
 	Created(Event) error
-	// Updating called when the model is being updated.
-	Updating(Event) error
 	// Updated called when the model has been updated.
 	Updated(Event) error
-	// Saving called when the model is being saved.
-	Saving(Event) error
-	// Saved called when the model has been saved.
-	Saved(Event) error
-	// Deleting called when the model is being deleted.
-	Deleting(Event) error
 	// Deleted called when the model has been deleted.
 	Deleted(Event) error
-	// ForceDeleting called when the model is being force deleted.
-	ForceDeleting(Event) error
 	// ForceDeleted called when the model has been force deleted.
 	ForceDeleted(Event) error
 }
 
+type ObserverWithCreating interface {
+	// Creating called when the model is being created.
+	Creating(Event) error
+}
+
+type ObserverWithDeleting interface {
+	// Deleting called when the model is being deleted.
+	Deleting(Event) error
+}
+
+type ObserverWithForceDeleting interface {
+	// ForceDeleting called when the model is being force deleted.
+	ForceDeleting(Event) error
+}
+
+type ObserverWithRestored interface {
+	// Restored called when the model has been restored.
+	Restored(Event) error
+}
+
+type ObserverWithRestoring interface {
+	// Restoring called when the model is being restored.
+	Restoring(Event) error
+}
+
+type ObserverWithRetrieved interface {
+	// Retrieved called when the model is retrieved from the database.
+	Retrieved(Event) error
+}
+
+type ObserverWithSaved interface {
+	// Saved called when the model has been saved.
+	Saved(Event) error
+}
+
+type ObserverWithSaving interface {
+	// Saving called when the model is being saved.
+	Saving(Event) error
+}
+
+type ObserverWithUpdating interface {
```

Comment:
Don't initiate these methods in the observer file. Add them according to requirement.